executor/table_reader.go

```diff
@@ -157,6 +163,27 @@ func (e *TableReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error
 		e.feedback.Invalidate()
 		return err
 	}
+
+	virCols := chunk.NewChunkWithCapacity(e.virtualColumnRetFieldTypes, req.Capacity())
+	iter := chunk.NewIterator4Chunk(req)
+
+	for i, idx := range e.virtualColumnIndex {
+		for row := iter.Begin(); row != iter.End(); row = iter.Next() {
+			datum, err := e.schema.Columns[idx].EvalVirtualColumn(row)
```

Comment:
`EvalVirtualDatum` converts the result to `Datum`, which is not efficient

Reply:
To use `CastValue` ,we have to get `datum`